app/services/registration/policy_engine.rb

```diff
@@ -0,0 +1,54 @@
+module Registration
+  class PolicyEngine
```

Comment:
A few sentences as class docstring would be nice to have, describing the overall motivation for this class without going into too much detail.